### ISSUE TITLE
Pi wakes up irregularly even when wakeup alarm is set to wakeup once a day

### ISSUE THREAD

**Issue by vvaibhav08** [PiSupply/PiJuice#881]
Hi! I have a Pi4 with a PiJuice module and a 12000mAh LiPo battery. I also have a solar panel connected. The entire setup is placed in a rather remote region. Now I have the Pi set up such that it should wake up once a day, run a process (take an image and upload to a cloud bucket), and then shut down, turning off the 5V as well. The setup wakes up as per the alarm specifications, but sometimes randomly wakes up at completely different times. The general pattern I've seen is that it would wake up randomly, triggering the process that I want to run on boot, shutdown, and then wake up multiple times quickly afterward in succession (as an example, I saw it woke up and uploaded an image 6 times in an hour period). I cannot figure out why that might be happening. 

Attached are the screenshots of the PiJuice configuration that I am using.
Setup:
Raspberry-pi4, 8GB
PiJuice, pijuice firmware v1.5
Battery: Pisupply, 12000mAh LiPo
Solar panel, 6W

General settings:
![Screenshot from 2022-06-23 10-39-17](https://user-images.githubusercontent.com/20376306/175258349-62ec95af-e99a-4d68-a49f-f2fc948925d1.png)

Wakeup alarm (I've also added `dtoverlay=i2c-rtc,ds1307`, in the boot config to make sure alarm flag remains checked)
![Screenshot from 2022-06-23 10-43-29](https://user-images.githubusercontent.com/20376306/175258470-4125fa57-4849-4b94-9c0b-f9a0beb3e304.png)

System task:
![Screenshot from 2022-06-23 10-43-59](https://user-images.githubusercontent.com/20376306/175258812-6f0bca15-d66f-4121-a7b4-455588da1d2c.png)

System events:
![Screenshot from 2022-06-23 10-44-14](https://user-images.githubusercontent.com/20376306/175258878-8888573e-c64d-45e3-bc25-5a85819a0463.png)

Would really appreciate some help. Cheers!

**Comment by tvoverbeek** [comment]
A 5W solar panel as (only) power source is probably too low.
The charging current for the 12000mAh is 2.5A.
A pi4 is also relatively power hungry.
My guess is that the battery goes so low that the MCU turns off. In that case the RTC loses its time until it gets resynchronizes when there is an internet connection (sysetemd_timesyncd).
Suggest you log the battery level at start to confirm that the battery has sufficient capacity.

**Comment by vvaibhav08** [comment]
@tvoverbeek thanks for the response. That is a good point and suggestion. However, it's not clear to me why this will cause the PiJuice to trigger a wake-up. Do you mean that after the MCU turns off, and the RTC clock loses time due to very low battery, the MCU will trigger an automatic wake-up the pi as soon as it detects some charging (from the solar panel)? 
In that case, would the solution be to just not let the battery get so low, and enable the "min-charge" setting?

**Comment by maxsitt** [comment]
@tvoverbeek @vvaibhav08 I have a very similar problem regarding irregular wakeup times.

Setup:
- Raspberry Pi Zero 2 W
- Pijuice Zero (Firmware v1.6)
- 12,000 mAh Pijuice battery
- solar panel charges powerbank, which is connected to the PiJuice

In my case there are several wakeup times every day:

![Screenshot (78)](https://user-images.githubusercontent.com/87697856/186872042-8d7e237a-b310-46fc-9a25-1aed7f027331.png)

While the wakeup is correctly triggered at the defined hours, it is additionally triggered about every 1-2 hours at irregular times.
I also changed "Weekday" to "Day", which seems to make no difference.

- No battery turn on is disabled
- PiJuice battery chargelevel is always >70%
- no internet connection, so no possible re-synchronisation of the time
- `dtoverlay=i2c-rtc,ds1307=1` in /boot/config.txt

Are there any other options I could try out to solve this problem? What could be possible causes despite power outage?

Thank you!

**Comment by tvoverbeek** [comment]
Note the alarm times are normally in UTC.
What is your timezone?
I am in Austria (now summertime: UTC+2)
So for me your daily settings are 4; 5;6;12;14; local time.

**Comment by maxsitt** [comment]
Yes, the times are already in UTC. I'm in the same timezone as you, so the actual wakeup times are 8;9;10;16;18;19.

However the RPi gets wake up a lot more, during the whole night every 1-2 hours and during the rest of the day between the defined hours.

**Comment by tvoverbeek** [comment]
I do not know how you charge the power bank by the solar array.
When the Pi is shut down do you remove the 5V from the PiJuice to the Pi?
In that case the residual power consumption by the PiJuice/Pi might be too low for the power bank.
Most power banks are designed for phone charging and turn off when the currrent drops below a certain value.
Maybe your solar array/power bank restarts and again delivers power to the Pi -> boot.

**Comment by maxsitt** [comment]
I'm using a [powerbank](https://voltaicsystems.com/v50/) with "Always on" feature, so that should be no problem if it works as advertised.

I activated `Software Halt Power Off` with a delay period of 65 seconds, so 5V to the Pi should be removed after about one minute after shutdown.

If the powerbank should restart anyway, will this power on the Pi even if `No battery turn on` is deactivated?
So if the USB micro power input changes from `NOT_PRESENT` to `PRESENT` the Pi will get turned on no matter what?

**Comment by tvoverbeek** [comment]
What is your 'Wakeup on charge' setting?
If it is enabled and the PiJuice battery level is above threshold the Pi will start as soon as the USB micro power input changes to PRESENT.

**Comment by maxsitt** [comment]
Wakeup on charge is disabled, as I only want the Pi to wake up at specific times.

The only other enabled System Task is Min Charge (Threshold 5%).

**Comment by tvoverbeek** [comment]
Then I am out of suggestions. Note (but I guess you are aware of this) 'Wakeup on charge' is not part of the System Task. There is also an additional non-volatile setting since the last firmware release.

**Comment by maxsitt** [comment]
Ok, thank you anyway for the help! I will try a different powerbank (or the solar panel as direct input into the PiJuice), as well as a normal power adapter and see what happens.

**Comment by maxsitt** [comment]
@tvoverbeek  I found the cause of the described problem! Now it seems kind of obvious, but I didn't think of it earlier, only after checking some of the settings. Even though the Watchdog setting was disabled, the Restore option on the right was enabled, which seems to activate the Watchdog even if it was disabled in the settings. I'm not sure if this is only happening on my system or if this could be improved by making the Restore option only available/active if the Watchdog setting is actually enabled.

So the Wakeup times were triggered by the Watchdog every 65min after the last shutdown. I can now see this pattern in the Wakeup times.

Unfortunately there is no documentation on the Restore setting and it is a little bit unintuitive that the Restore setting activates the Watchdog even if the Watchdog was disabled by the user.

**Comment by jei-gaither** [comment]
Howdy! I've been listening to this thread and this explains my issue as well!

I had my Watchdog set to ON (but not Restore), and since your post pointed it out, the pattern is obvious to me as well.

Two things:

A) Can you shed more light on what Restore setting is designed for in terms of Watchdog and Wake Up on Charge? It's unclear to me what Restore setting does.

and

2) Is there a workaround for using Watchdog (in case of lockups) and NOT having it trigger a wake state?

Thanks for this awesome product and the great efforts to help other users/developers!!!

   -Jei.

**Comment by vvaibhav08** [comment]
@maxsitt, great that you found the issue for your repeated unwanted wake-ups. I am not really sure why I still had these wake-ups even when both the watchdog and the restore settings were disabled in my setup. It's so strange. Anyways, since I never could find the root cause for this, I shamelessly just updated my boot script to check if the current time matches the intended wakeup alarm times, run the process if it does, and shut down otherwise. Quite dirty but works for me.
